### PR TITLE
client-go/events: use the predefined error judgment

### PR DESCRIPTION
**Pull request by zc2638**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
```
The `reference.GetReference` method has judged the case where the `related` is nil, and we should use the defined error to judge.
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/events/event_recorder.go [hunk 1]
```diff
@@ -49,13 +49,11 @@ func (recorder *recorderImpl) Eventf(regarding runtime.Object, related runtime.O
...
 	}
 
-	var refRelated *v1.ObjectReference
-	if related != nil {
-		refRelated, err = reference.GetReference(recorder.scheme, related)
-		if err != nil {
-			klog.V(9).Infof("Could not construct reference to: '%#v' due to: '%v'.", related, err)
-		}
+	refRelated, err := reference.GetReference(recorder.scheme, related)
+	if err != nil && err != reference.ErrNilObject {
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/tools/events/event_recorder.go:53]: i feel the original code is easier to read

**Comment by zc2638** [comment]
/cc @caesarxuchao 
/cc @yliaog 
Thank you for your review, are there any more questions here, I can deal with it.

**Comment by yliaog** [comment]
originally, it is clear that it is checking related != nil, after the change, from err != reference.ErrNilObject, it is not clear what is ErrNilObject, reader has to go to the function reference.GetReference to find out.

**Comment by zc2638** [comment]
> originally, it is clear that it is checking related != nil, after the change, from err != reference.ErrNilObject, it is not clear what is ErrNilObject, reader has to go to the function reference.GetReference to find out.

Thanks for your opinion.
I have an example for your reference.

When we use client-go, we often use `apierrors.IsNotFound(err)` or `ReasonForError(err) == metav1.StatusReasonNotFound` to make some judge. And why don’t we check if the variable is nil.

And my little insight, I think this error variable name `reference.ErrNilObject` is clear enough to describe what the error is.
It can be recognized by reading the variable name directly, it should be an empty object error.

So, I’m not sure what you’re worried about?

**Comment by dims** [comment]
@zc2638 feel free to look in the OWNERS file to see if anyone else agrees/dis-agrees. we can close this PR if you don't get any support

**Comment by zc2638** [comment]
Thanks, it's not some very important modification, I won't spend more time on it.
@dims

**Pull request closed without merging**